api/src/app/modules/project/project.service.ts

```diff
@@ -725,5 +750,53 @@ export class ProjectService extends DataService<Project, Repository<Project>, Pr
     this.logger.debug(`${updatedCounts} FOM(s) for ${projectIds} were updated to ${workflowStateCode}`);
   }
 
+  private async updatePublicNoticeIfRequired(projectEntity: Project, stateTransition: WorkflowStateEnum) {
+		if (WorkflowStateEnum.PUBLISHED == stateTransition) {
+			const publicNotices = projectEntity.publicNotices;
+			// set public notice post date to project commenting open date if post date is not set.
+			if (!_.isEmpty(publicNotices) && !publicNotices[0].postDate) {
+				const pnId = publicNotices[0].id;
+				const commentingOpenDate = projectEntity.commentingOpenDate;
+				const updateFields = 
+				{
+					postDate: commentingOpenDate,
+					updateUser: 'system',
+					updateTimestamp: new Date(),
+					revisionCount: () => "revision_count + 1"
+				}
+				const updatedCount = (await this.repository
```

Comment:
Why are we doing this?

Reply:
At the project service, during the workflow_state_change to "PUBLISH", this is where the code update the publicNotice entity post_date to set to commenting_open_date if post_date is empty.